From 561fe3cb65e01ba60d0c59c698aa6efc87add008 Mon Sep 17 00:00:00 2001
From: Chung-Hsien Hsu <stanley.hsu@cypress.com>
Date: Mon, 5 Oct 2020 04:15:51 -0500
Subject: [PATCH 131/138] brcmfmac: use firmware_request_nowarn for the
 board-specific nvram

Loading a board-specific nvram file is optional. If the file is not
present, we will get a warning "Direct firmware load for ... failed with
error ..." and fall back to a common one. However, users may treat the
warning as some type of side effects and the file is required.

This commit switches to firmware_request_nowarn() and leaves an info
message in place to avoid confusion.

Signed-off-by: Chung-Hsien Hsu <stanley.hsu@cypress.com>
Signed-off-by: Chi-hsien Lin <chi-hsien.lin@infineon.com>
---
 drivers/net/wireless/broadcom/brcm80211/brcmfmac/firmware.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/firmware.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/firmware.c
index 3aed4c4..33bfcdb 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/firmware.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/firmware.c
@@ -613,8 +613,11 @@ static int brcmf_fw_request_firmware(const struct firmware **fw,
 		strlcat(alt_path, fwctx->req->board_type, BRCMF_FW_NAME_LEN);
 		strlcat(alt_path, ".txt", BRCMF_FW_NAME_LEN);
 
-		ret = request_firmware(fw, alt_path, fwctx->dev);
-		if (ret == 0)
+		ret = firmware_request_nowarn(fw, alt_path, fwctx->dev);
+		if (ret)
+			brcmf_info("no board-specific nvram available (err=%d), device will use %s\n",
+				   ret, cur->path);
+		else
 			return ret;
 	}
 
-- 
1.9.1

